### PR TITLE
don't block main thread during search

### PR DESCRIPTION
**Pull request by michaelkirk**
The search doesn't take too long, but there's a very perceptible momentary hang while typing in your search term.

I don't think it's slow enough to introduce a progress indicator, in fact I think doing so would be more damaging than helpful since it would blip in and out.

PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift [hunk 1]
```diff
@@ -378,12 +378,19 @@ class ConversationSearchViewController: UITableViewController {
...
 
-        self.uiDatabaseConnection.read { transaction in
-            self.searchResultSet = self.searcher.results(searchText: searchText, transaction: transaction, contactsManager: self.contactsManager)
-        }
-
-        // TODO: more performant way to do this?
-        self.tableView.reloadData()
+        var searchResults: SearchResultSet?
+        self.uiDatabaseConnection.asyncRead({ transaction in
+            searchResults = self.searcher.results(searchText: searchText, transaction: transaction, contactsManager: self.contactsManager)
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift]: Should we do a `weakSelf` thing here?

File: Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift [hunk 2]
```diff
@@ -378,12 +378,19 @@ class ConversationSearchViewController: UITableViewController {
...
+        },
+                                             completionBlock: {
+                                                guard let results = searchResults else {
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift]: And again, separately, here?

Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift]: let's assert that we're on the main thread here?

**Comment by michaelkirk-signal** [comment]
In aabf9e79e

**Pull request closed without merging**